In [1]:
import sys

In [2]:
sys.path.append(r'C:\Users\jk\Desktop\data science - infoshare\projekt_SQL\jdszr4-edc\2-projekt-python/misc')

In [3]:
from matplotlib import pyplot
import pandas as pd
import numpy as np

In [4]:
import data_preparation as prep

In [5]:
df = prep.df
df

Age  Attrition  BusinessTravel  DailyRate  Department  DistanceFromHome  \
0      41          1               1       1102           2                 1   
1      49          0               2        279           1                 8   
2      37          1               1       1373           1                 2   
3      33          0               2       1392           1                 3   
4      27          0               1        591           1                 2   
...   ...        ...             ...        ...         ...               ...   
1465   36          0               2        884           1                23   
1466   39          0               1        613           1                 6   
1467   27          0               1        155           1                 4   
1468   49          0               2       1023           2                 2   
1469   34          0               1        628           1                 8   

      Education  EducationField  EnvironmentSatisfaction  Gender  ...  \
0             2               1                        2       0  ...   
1             1               1                        3       1  ...   
2             2               5                        4       1  ...   
3             4               1                        4       0  ...   
4             1               3                        1       1  ...   
...         ...             ...                      ...     ...  ...   
1465          2               3                        3       1  ...   
1466          1               3                        4       1  ...   
1467          3               1                        2       1  ...   
1468          3               3                        4       1  ...   
1469          3               3                        2       1  ...   

      PerformanceRating  RelationshipSatisfaction  StockOptionLevel  \
0                     3                         1                 0   
1                     4                         4                 1   
2                     3                         2                 0   
3                     3                         3                 0   
4                     3                         4                 1   
...                 ...                       ...               ...   
1465                  3                         3                 1   
1466                  3                         1                 1   
1467                  4                         2                 1   
1468                  3                         4                 0   
1469                  3                         1                 0   

      TotalWorkingYears  TrainingTimesLastYear  WorkLifeBalance  \
0                     8                      0                1   
1                    10                      3                3   
2                     7                      3                3   
3                     8                      3                3   
4                     6                      3                3   
...                 ...                    ...              ...   
1465                 17                      3                3   
1466                  9                      5                3   
1467                  6                      0                3   
1468                 17                      3                2   
1469                  6                      3                4   

      YearsAtCompany  YearsInCurrentRole  YearsSinceLastPromotion  \
0                  6                   4                        0   
1                 10                   7                        1   
2                  0                   0                        0   
3                  8                   7                        3   
4                  2                   2                        2   
...              ...                 ...                      ...   
1465           

# Analiza poziomu satysfakcji pod względem całkowitej liczby lat doświadczenia - total working years

In [6]:
df_TotalWorkingYears = df[['TotalWorkingYears', 'EnvironmentSatisfaction', 'JobSatisfaction', 'RelationshipSatisfaction']]
df_TotalWorkingYears.sort_values(by=['TotalWorkingYears'])

TotalWorkingYears  EnvironmentSatisfaction  JobSatisfaction  \
1153                  0                        2                4   
1311                  0                        2                3   
457                   0                        2                2   
828                   0                        3                3   
615                   0                        4                4   
...                 ...                      ...              ...   
85                   37                        4                4   
1301                 37                        2                2   
98                   38                        4                3   
595                  40                        4                2   
126                  40                        4                4   

      RelationshipSatisfaction  
1153                         3  
1311                         3  
457                          4  
828                          4  
615                          3  
...                        ...  
85                           1  
1301                         4  
98                           3  
595                          4  
126                          4  

[1470 rows x 4 columns]

In [7]:
#sprawdzenie danych dla TotalWorkingYears

In [8]:
df.describe()['TotalWorkingYears']

count    1470.000000
mean       11.279592
std         7.780782
min         0.000000
25%         6.000000
50%        10.000000
75%        15.000000
max        40.000000
Name: TotalWorkingYears, dtype: float64

In [9]:
df_25 = df['TotalWorkingYears'].quantile(0.25)
df_50 = df['TotalWorkingYears'].quantile(0.5)
df_75 = df['TotalWorkingYears'].quantile(0.75)
df_90 = df['TotalWorkingYears'].quantile(0.9)
print(df_25, df_50, df_75, df_90)

6.0 10.0 15.0 23.0


In [10]:
#na podstawie wyliczonych kwantyli, wyznaczam przedziały, dla których będę przeprowadzać analizę
#T1 - osoby, które pracują 6 lat lub krócej
#T2 - osoby, które pracują dłużej niż 6 lat, ale krócej niż 10
#T3 - osoby, które pracują dłużej niż 10 lat, ale krócej niż 15
#T4 - osoby, które pracują dłużej niż 15 lat, ale krócej niż 23
#T5 - osoby, które pracują dłużej niż 23 lata

In [78]:
#Ogólna statystyka dla całego przedziału TotalWorkingYears
T = df['TotalWorkingYears']
T_enivronment = df['EnvironmentSatisfaction'].mean()
T_job = df['JobSatisfaction'].mean()
T_relationship = df['RelationshipSatisfaction'].mean()
T_overall = (T_enivronment + T_job +T_relationship)/3

print(T_enivronment,T_job, T_relationship, T_overall)

2.721768707482993 2.7285714285714286 2.7122448979591836 2.7208616780045354


In [11]:
T1 = df[(df.TotalWorkingYears <= 6)]
T1_enivronment = T1['EnvironmentSatisfaction'].mean()
T1_job =T1['JobSatisfaction'].mean()
T1_relationship = T1['RelationshipSatisfaction'].mean()
T1_overall = (T1_enivronment + T1_job +T1_relationship)/3

print(T1_enivronment,T1_job, T1_relationship, T1_overall)

2.7165532879818595 2.7301587301587302 2.7346938775510203 2.7271352985638697


In [51]:
T2 = df[(df.TotalWorkingYears > 6) & (df.TotalWorkingYears <= 10)]
T2_enivronment = T2['EnvironmentSatisfaction'].mean()
T2_job = T2['JobSatisfaction'].mean()
T2_relationship = T2['RelationshipSatisfaction'].mean()
T2_overall = (T2_enivronment + T2_job +T2_relationship)/3

print(T2_enivronment,T2_job, T2_relationship, T2_overall)

2.711618257261411 2.7427385892116183 2.6825726141078836 2.7123098201936373


In [52]:
T3 = df[(df.TotalWorkingYears > 10) & (df.TotalWorkingYears <= 15)]
T3_enivronment = T3['EnvironmentSatisfaction'].mean()
T3_job = T3['JobSatisfaction'].mean()
T3_relationship = T3['RelationshipSatisfaction'].mean()
T3_overall = (T3_enivronment + T3_job +T3_relationship)/3

print(T3_enivronment,T3_job, T3_relationship, T3_overall)

2.738219895287958 2.816753926701571 2.5968586387434556 2.717277486910995


In [83]:
T4 = df[(df.TotalWorkingYears > 15) & (df.TotalWorkingYears <= 23)]
T4_enivronment = T4['EnvironmentSatisfaction'].mean()
T4_job = T4['JobSatisfaction'].mean()
T4_relationship = T4['RelationshipSatisfaction'].mean()
T4_overall = (T4_enivronment + T4_job +T4_relationship)/3

print(T4_enivronment,T4_job, T4_relationship, T4_overall)

2.747787610619469 2.6238938053097347 2.774336283185841 2.7153392330383483


Age  Attrition  BusinessTravel  DailyRate  Department  DistanceFromHome  \
9      36          0               1       1299           1                27   
29     46          0               1        705           2                 2   
33     39          1               1        895           2                 5   
45     41          1               1       1360           1                12   
50     48          1               1        626           1                 1   
...   ...        ...             ...        ...         ...               ...   
1457   40          0               1       1194           1                 2   
1461   50          1               1        410           2                28   
1462   39          0               1        722           2                24   
1465   36          0               2        884           1                23   
1468   49          0               2       1023           2                 2   

      Education  EducationField  EnvironmentSatisfaction  Gender  ...  \
9             3               3                        3       1  ...   
29            4               2                        2       0  ...   
33            3               4                        4       1  ...   
45            3               4                        2       0  ...   
50            2               1                        1       1  ...   
...         ...             ...                      ...     ...  ...   
1457          4               3                        3       0  ...   
1461          3               2                        4       1  ...   
1462          1               2                        2       0  ...   
1465          2               3                        3       1  ...   
1468          3               3                        4       1  ...   

      PerformanceRating  RelationshipSatisfaction  StockOptionLevel  \
9                     3                         2                 2   
29                    3                         4                 0   
33                    3                         3                 1   
45                    3                         4                 0   
50                    3                         4                 0   
...                 ...                       ...               ...   
1457                  3                         2                 3   
1461                  3                         2                 1   
1462                  3                         1                 1   
1465                  3                         3                 1   
1468                  3                         4                 0   

      TotalWorkingYears  TrainingTimesLastYear  WorkLifeBalance  \
9                    17                      3                2   
29                   22                      2                2   
33                   19                      6                4   
45                   23                      0                3   
50                   23                      2                3   
...                 ...                    ...              ...   
1457                 20                      2                3   
1461                 20                      3                3   
1462                 21                      2                2   
1465                 17                      3                3   
1468                 17                      3                2   

      YearsAtCompany  YearsInCurrentRole  YearsSinceLastPromotion  \
9                  7                   7                        7   
29                 2                   2                        2   
33                 1                   0                        0   
45                22                  15                       15   
50                 1                   0                        0   
...              ...                 ...                      ...   
1457           

In [54]:
T5 = df[df.TotalWorkingYears > 23]
T5_enivronment = T5['EnvironmentSatisfaction'].mean()
T5_job = T5['JobSatisfaction'].mean()
T5_relationship = T5['RelationshipSatisfaction'].mean()
T5_overall = (T5_enivronment + T5_job +T5_relationship)/3

print(T5_enivronment,T5_job, T5_relationship, T5_overall)

2.707692307692308 2.723076923076923 2.8076923076923075 2.7461538461538466


|_PRZEDZIAŁ CAŁKOWITEGO DOŚWIADCZENIA_ ||_ŚREDNIA SATYSFAKCJA ZE ŚRODOWISKA PRACY_ ||_ŚREDNIA SATYSFAKCJA Z PRACY_||_ŚREDNIA SATYSFAKCJA Z RELACJI W PRACY_||_ŚREDNI CAŁKOWITY POZIOM SATYSFAKCJI|
|-----||-----||-----||-----||-----|
| <6||2.72||2.73||2.73||2.73|
|6 - 10||2.71||2.74||2.68||2.71|
|10 - 15||2.74||2.82||2.60||2.72|
|15 - 23||2.75||2.62||2.77||2.72|
|>23||2.71||2.72||2.81||2.75|

In [16]:
def corr_color(val):
    if abs(val) == 1.0:
        color = 'grey'
    elif abs(val) > 0.8:
        color = 'red'
    elif abs(val) > 0.5:
        color = 'orange'
    elif abs(val) > 0.3:
        color = 'yellow'
    else:
        color = 'blue'
    return 'color: %s' % color

In [79]:
#Ogólne korelacje dla cąłego przedziału TotalWorkingYears
T1_corr = df.corr()
T1_corr = T1_corr.style.applymap(corr_color)
T1_corr

Nie znaleziono korelacji pomiędzy całkowitym sta zem pracy, aktórymkolwiek z badanych czynników satysfakcji.

Dodatkowo: korelacja z wiekiem, poziomem stanowiska, miesięcznym przychodem, stażem w firmie, stażem na obecnym stanowisku, stażem z obecnym managerem oraz czasem od ostatniego awansu.

In [17]:
T1_corr = T1.corr()
T1_corr = T1_corr.style.applymap(corr_color)
T1_corr

Wśród gurpy osób pracujących krócej niż 6 lat, nie ma korelacji pomiędzy stażem pracy a którymkolwiek z badanych czynników satysfakcji.

Dodatkowo: korelacja z poziomem stanowiska, z miesięcznymi dochodami, ze stażem w firmie, stażem na obecnym stanowisku, oraz stażem z obecnym managerem.

In [18]:
T2_corr = T2.corr()
T2_corr = T2_corr.style.applymap(corr_color)
T2_corr

Wśród gurpy osób pracujących dłużej niż 6 lat, lecz krócej niż 10, nie ma korelacji pomiędzy stażem pracy a którymkolwiek z badanych czynników satysfakcji

Dodatkow: dodatnia korelacja z poziomem stanowiska, miesięcznym przychodem, stażem w firmie oraz z obecnym managerem.

In [19]:
T3_corr = T3.corr()
T3_corr = T3_corr.style.applymap(corr_color)
T3_corr

Wśród gurpy osób pracujących dłużej niż 10 lat, lecz krócej niż 15, nie ma korelacji pomiędzy stażem pracy a którymkolwiek z badanych czynników satysfakcji.

Brak dodatkowych korelacji.

In [20]:
T4_corr = T4.corr()
T4_corr = T4_corr.style.applymap(corr_color)
T4_corr

Wśród gurpy osób pracujących dłużej niż 15 lat lecz krócej niż 23 lata, nie ma korelacji pomiędzy stażem pracy a którymkolwiek z badanych czynników satysfakcji.

Doatkowo: korelacja z wiekiem, poziomem stanowiska, miesięcznym przychodem.

In [21]:
T5_corr = T5.corr()
T5_corr = T5_corr.style.applymap(corr_color)
T5_corr

Wśród gurpy osób pracujących dłużej niż 23 lata, nie ma korelacji pomiędzy stażem pracy a którymkolwiek z badanych czynników satysfakcji

Dodatkowo: korelacja z wiekiem, poziomem edukacji.

# Analiza poziomu satysfakcji pod względem liczby lat doświadczenia w firmie - years at company

In [22]:
df_YearsAtCompany = df[['YearsAtCompany', 'EnvironmentSatisfaction', 'JobSatisfaction', 'RelationshipSatisfaction']]
df_YearsAtCompany.sort_values(by=['YearsAtCompany'])

YearsAtCompany  EnvironmentSatisfaction  JobSatisfaction  \
1311               0                        2                3   
1371               0                        4                1   
460                0                        1                3   
1068               0                        3                1   
457                0                        2                2   
...              ...                      ...              ...   
561               34                        3                1   
1116              36                        3                4   
270               36                        4                1   
98                37                        4                3   
126               40                        4                4   

      RelationshipSatisfaction  
1311                         3  
1371                         3  
460                          1  
1068                         3  
457                          4  
...                        ...  
561                          1  
1116                         3  
270                          3  
98                           3  
126                          4  

[1470 rows x 4 columns]

In [23]:
df.describe()['YearsAtCompany']

count    1470.000000
mean        7.008163
std         6.126525
min         0.000000
25%         3.000000
50%         5.000000
75%         9.000000
max        40.000000
Name: YearsAtCompany, dtype: float64

In [24]:
dfy_25 = df['YearsAtCompany'].quantile(0.25)
dfy_50 = df['YearsAtCompany'].quantile(0.5)
dfy_75 = df['YearsAtCompany'].quantile(0.75)
dfy_90 = df['YearsAtCompany'].quantile(0.9)
print(dfy_25, dfy_50, dfy_75, dfy_90)

3.0 5.0 9.0 15.0


In [25]:
#na podstawie wyliczonych kwantyli, wyznaczam przedziały, dla których będę przeprowadzać analizę
#Y1 - osoby, które pracują w firmie 3 lata lub krócej
#Y2 - osoby, które pracują w firmie dłużej niż 3 lata, ale krócej niż 5
#Y3 - osoby, które pracują w firmie dłużej niż 5 lat, ale krócej niż 9
#Y4 - osoby, które pracują w firmie dłużej niż 9 lat,a le krócej niż 15
#Y5 - osoby, które pracują w firmie dłużej niż 15 lat

In [71]:
#Ogólna statystyka dla całego przedziału YearsAtCompany
Y = df['YearsAtCompany']
Y_enivronment = df['EnvironmentSatisfaction'].mean()
Y_job =df['JobSatisfaction'].mean()
Y_relationship = df['RelationshipSatisfaction'].mean()
Y_overall = (Y_enivronment+Y_job+Y_relationship)/3

print(Y_enivronment,Y_job, Y_relationship, Y_overall)

2.721768707482993 2.7285714285714286 2.7122448979591836 2.7208616780045354


In [56]:
Y1 = df[df.YearsAtCompany <= 3]
Y1_enivronment = Y1['EnvironmentSatisfaction'].mean()
Y1_job =Y1['JobSatisfaction'].mean()
Y1_relationship = Y1['RelationshipSatisfaction'].mean()
Y1_overall = (Y1_enivronment+Y1_job+Y1_relationship)/3

print(Y1_enivronment,Y1_job, Y1_relationship, Y1_overall)

2.6702127659574466 2.7148936170212767 2.74468085106383 2.7099290780141843


In [57]:
Y2 = df[(df.YearsAtCompany > 3) & (df.YearsAtCompany <= 5)]
Y2_enivronment = Y2['EnvironmentSatisfaction'].mean()
Y2_job =Y2['JobSatisfaction'].mean()
Y2_relationship = Y2['RelationshipSatisfaction'].mean()
Y2_overall = (Y2_enivronment+Y2_job+Y2_relationship)/3

print(Y2_enivronment,Y2_job, Y2_relationship, Y2_overall)

2.735294117647059 2.689542483660131 2.7058823529411766 2.710239651416122


In [58]:
Y3 = df[(df.YearsAtCompany > 5) & (df.YearsAtCompany <= 9)]
Y3_enivronment = Y3['EnvironmentSatisfaction'].mean()
Y3_job =Y3['JobSatisfaction'].mean()
Y3_relationship = Y3['RelationshipSatisfaction'].mean()
Y3_overall = (Y3_enivronment+Y3_job+Y3_relationship)/3

print(Y3_enivronment,Y3_job, Y3_relationship, Y3_overall)

2.8292682926829267 2.7560975609756095 2.652439024390244 2.7459349593495936


In [59]:
Y4 = df[(df.YearsAtCompany > 9) & (df.YearsAtCompany <= 15)]
Y4_enivronment = Y4['EnvironmentSatisfaction'].mean()
Y4_job =Y4['JobSatisfaction'].mean()
Y4_relationship = Y4['RelationshipSatisfaction'].mean()
Y4_overall = (Y4_enivronment+Y4_job+Y4_relationship)/3

print(Y4_enivronment,Y4_job, Y4_relationship, Y4_overall)

2.6973684210526314 2.8157894736842106 2.675438596491228 2.72953216374269


In [60]:
Y5 = df[df.YearsAtCompany > 15]
Y5_enivronment = Y4['EnvironmentSatisfaction'].mean()
Y5_job =Y5['JobSatisfaction'].mean()
Y5_relationship = Y5['RelationshipSatisfaction'].mean()
Y5_overall = (Y5_enivronment+Y5_job+Y5_relationship)/3

print(Y5_enivronment,Y5_job, Y5_relationship, Y5_overall)

2.6973684210526314 2.652173913043478 2.818840579710145 2.722794304602085


|_PRZEDZIAŁ DOŚWIADCZENIA W FIRMIE_ ||_ŚREDNIA SATYSFAKCJA ZE ŚRODOWISKA PRACY_ ||_ŚREDNIA SATYSFAKCJA Z PRACY_||_ŚREDNIA SATYSFAKCJA Z RELACJI W PRACY_||_ŚREDNI CAŁKOWITY POZIOM SATYSFAKCJI|
|-----||-----||-----||-----||-----|
| <3||2.67||2.71||2.74||2.71|
|3 - 5||2.74||2.69||2.71||2.71|
|5 - 9||2.83||2.76||2.65||2.75|
|9 - 15||2.70||2.82||2.68||2.73|
|>15||2.70||2.65||2.82||2.72|

In [75]:
#Ogólne korelacje dla przediału YearsAtCompany
Y_corr = df.corr()
Y_corr = Y_corr.style.applymap(corr_color)
Y_corr

Nie znaleziono korelacji pomiędzy całkowitym stażem pracy w firmie, a badanymi czynnikami satysfakcji.

Dodatkowo: korelacja z wiekiem, poziomem stanowiska, miesięcznym przychodem, całkowitym stażem pracy, stażem na obcenym stanowisku, czasem od ostatniego awansu oraz stażem z obecnym managerem.

In [73]:
Y1_corr = Y1.corr()
Y1_corr = Y1_corr.style.applymap(corr_color)
Y1_corr

Wśród osób pracujących w firmie krócej niż 3 lata nie ma korelacji pomiędzy stażem, a badanymi czynnikami satysfakcji. 

Dodatkowo: silnie dodatnie korelacje pomiędzy stażem w firmie, a stażem na danym stanowisku, czasem od ostatniego awansu oraz ze stażem pracy z obecnym managerem.

In [32]:
Y2_corr = Y2.corr()
Y2_corr = Y2_corr.style.applymap(corr_color)
Y2_corr

Wśród osób pracujących w firmie dłużej niż 3 lata lecz krócej lub równo 5 lat nie ma korelacji pomiędzy stażem, a badanymi czynnikami satysfakcji. 

Brak dodatkowych korelacji.

In [33]:
Y3_corr = Y3.corr()
Y3_corr = Y3_corr.style.applymap(corr_color)
Y3_corr

Wśród osób pracujących w firmie dłużej niż 5 lat lecz krócej niż 9 nie ma korelacji pomiędzy stażem, a badanymi czynnikami satysfakcji. 

Dodatkowo: dodtanie korelacje pomiędzy stażem w firmie, a stażem na obecnym stanowisku oraz z obecnym managerem.

In [34]:
Y4_corr = Y4.corr()
Y4_corr = Y4_corr.style.applymap(corr_color)
Y4_corr

Wśród osób pracujących w firmie dłużej niż 9 lat lecz krócej niż 15 nie ma korelacji pomiędzy stażem, a badanymi czynnikami satysfakcji. 

Dodatkowo: dodatanie korelacje pomiędzy stażem w firmie, a całkowitym stażem pracy, stażem na obecnym stanowisku oraz stażem z obecnym managerem.

In [35]:
Y5_corr = Y5.corr()
Y5_corr = Y5_corr.style.applymap(corr_color)
Y5_corr

Wśród osób pracujących w firmie dłużej niż 15 lat nie ma korelacji pomiędzy stażem, a badanymi czynnikami satysfakcji. 

Dodatkowo: dodatanie korelacje pomiędzy stażem w firmie, a wiekiem, poziomem stanowiska, miesięcznym przychodem oraz całkowitym stażem pracy.

# Analiza poziomu satysfakcji pod względem liczby lat doświadczenia na obecnym stanowisku - years at current role

In [36]:
df_YearsInCurrentRole = df[['YearsInCurrentRole', 'EnvironmentSatisfaction', 'JobSatisfaction', 'RelationshipSatisfaction']]
df_YearsInCurrentRole.sort_values(by=['YearsInCurrentRole'])

YearsInCurrentRole  EnvironmentSatisfaction  JobSatisfaction  \
742                    0                        1                4   
1233                   0                        2                4   
906                    0                        3                3   
1234                   0                        3                2   
258                    0                        3                4   
...                  ...                      ...              ...   
1327                  17                        1                1   
1351                  17                        4                4   
231                   17                        3                4   
123                   18                        1                3   
190                   18                        3                3   

      RelationshipSatisfaction  
742                          2  
1233                         2  
906                          4  
1234                         4  
258                          2  
...                        ...  
1327                         4  
1351                         2  
231                          4  
123                          3  
190                          1  

[1470 rows x 4 columns]

In [37]:
df.describe()['YearsInCurrentRole']

count    1470.000000
mean        4.229252
std         3.623137
min         0.000000
25%         2.000000
50%         3.000000
75%         7.000000
max        18.000000
Name: YearsInCurrentRole, dtype: float64

In [38]:
dfc_25 = df['YearsInCurrentRole'].quantile(0.25)
dfc_50 = df['YearsInCurrentRole'].quantile(0.5)
dfc_75 = df['YearsInCurrentRole'].quantile(0.75)
dfc_90 = df['YearsInCurrentRole'].quantile(0.9)
print(dfc_25, dfc_50, dfc_75, dfc_90)

2.0 3.0 7.0 9.0


In [39]:
#na podstawie wyliczonych kwantyli, wyznaczam przedziały, dla których będę przeprowadzać analizę
#C1 - osoby, które pracują na obecnym stanowisku 2 lata lub krócej
#C2 - osoby, które pracują na obecnym stanowisku dłużej niż 2 lata, ale krócej lub równo 3
#C3 - osoby, które pracują na obecnym stanowisku dłużej niż 3 lata, ale krócej lub równo 7
#C4 - osoby, które pracują na obecnym stanowisku dłużej niż 7 lata, ale krócej lub równo 9
#C5 - osoby, które pracują na obecnym stanowisku dłużej niż 9 lat

In [81]:
#ogólna statystyka dla przedziału YearsAtCurrentRole
C = df['YearsInCurrentRole']
C_enivronment = df['EnvironmentSatisfaction'].mean()
C_job = df['JobSatisfaction'].mean()
C_relationship = df['RelationshipSatisfaction'].mean()
C_overall = (C_enivronment + C_job + C_relationship)/3

print(C_enivronment,C_job, C_relationship, C_overall)

2.721768707482993 2.7285714285714286 2.7122448979591836 2.7208616780045354


In [62]:
C1 = df[df.YearsInCurrentRole <= 2]
C1_enivronment = C1['EnvironmentSatisfaction'].mean()
C1_job = C1['JobSatisfaction'].mean()
C1_relationship = C1['RelationshipSatisfaction'].mean()
C1_overall = (C1_enivronment + C1_job + C1_relationship)/3

print(C1_enivronment,C1_job, C1_relationship, C1_overall)

2.695393759286776 2.736998514115899 2.7726597325408617 2.7350173353145117


In [63]:
C2 = df[(df.YearsInCurrentRole > 2) & (df.YearsInCurrentRole <= 3)]
C2_enivronment = C2['EnvironmentSatisfaction'].mean()
C2_job = C2['JobSatisfaction'].mean()
C2_relationship = C2['RelationshipSatisfaction'].mean()
C2_overall = (C2_enivronment + C2_job + C2_relationship)/3

print(C2_enivronment,C2_job, C2_relationship, C2_overall)

2.688888888888889 2.6222222222222222 2.6222222222222222 2.6444444444444444


In [66]:
C3 = df[(df.YearsInCurrentRole > 3) & (df.YearsInCurrentRole <= 7)]
C3_environment = C3['EnvironmentSatisfaction'].mean()
C3_job = C3['JobSatisfaction'].mean()
C3_relationship = C3['RelationshipSatisfaction'].mean()
C3_overall = (C3_environment + C3_job + C3_relationship)/3

print(C3_environment,C3_job, C3_relationship, C3_overall)

2.7343358395989976 2.7593984962406015 2.6340852130325816 2.709273182957394


In [67]:
C4 = df[(df.YearsInCurrentRole > 7) & (df.YearsInCurrentRole <= 9)]
C4_environment = C3['EnvironmentSatisfaction'].mean()
C4_job = C4['JobSatisfaction'].mean()
C4_relationship = C4['RelationshipSatisfaction'].mean()
C4_overall = (C4_environment + C4_job + C4_relationship)/3

print(C4_environment,C4_job, C4_relationship, C4_overall)

2.7343358395989976 2.7051282051282053 2.6858974358974357 2.7084538268748797


In [68]:
C5 = df[df.YearsInCurrentRole > 9]
C5_environment = C5['EnvironmentSatisfaction'].mean()
C5_job = C5['JobSatisfaction'].mean()
C5_relationship = C5['RelationshipSatisfaction'].mean()
C5_overall = (C5_environment + C5_job + C5_relationship)/3

print(C5_environment,C5_job, C5_relationship, C5_overall)

2.7570093457943927 2.7289719626168223 2.7757009345794392 2.753894080996885


|_PRZEDZIAŁ DOŚWIADCZENIA NA OBECNYM STANOWISKU_ ||_ŚREDNIA SATYSFAKCJA ZE ŚRODOWISKA PRACY_ ||_ŚREDNIA SATYSFAKCJA Z PRACY_||_ŚREDNIA SATYSFAKCJA Z RELACJI W PRACY_||_ŚREDNI CAŁKOWITY POZIOM SATYSFAKCJI|
|-----||-----||-----||-----||-----|
| <2||2.70||2.73||2.77||2.74|
|2 - 3||2.69||2.62||2.62||2.64|
|3 - 5||2.73||2.76||2.63||2.71|
|5 - 9||2.73||2.73||2.69||2.71|
|>9||2.76||2.73||2.78||2.75|
|total||2.72||2.73||2.71||2.72|

In [82]:
#korelacje dla całego przedziału YearsAtCurrentRole
C_corr = df.corr()
C_corr = C_corr.style.applymap(corr_color)
C_corr

Brak korelacji pomiędzy stażem na obecnym stanowisku, a którymkolwiek z badanych czynników satysfakcji.

Dodatkow: korelacja z poziomem stanowiska, miesięcznym przychodem, całkowitym stażem pracy, stażem w firmie, stażem z obecnym managerem, czasem od ostatniego awansu.

In [45]:
C1_corr = C1.corr()
C1_corr = C1_corr.style.applymap(corr_color)
C1_corr

Wśród osób o stażu nn obecnym stanowisku krótszym niż 2 lata nie znaleziono korelacji pomiędzy stażem na obecnym stanowisku a badanymi czynnikami satysfakcji.

Dodatkowo: znaleziono korelację pomiędzy stażem na obecnym stanowisku a stażem z obecnym managerem.

In [46]:
C2_corr = C2.corr()
C2_corr = C2_corr.style.applymap(corr_color)
C2_corr

Wśród osób o stażu nn obecnym stanowisku krótszym lub równym 3 lata nie znaleziono korelacji pomiędzy stażem na obecnym stanowisku a badanymi czynnikami satysfakcji.

Dodatkowo: znaleziono korelację pomiędzy stażem na obecnym stanowisku a stażem z obecnym managerem.

In [48]:
C3_corr = C3.corr()
C3_corr = C3_corr.style.applymap(corr_color)
C3_corr

Wśród osób o stażu nn obecnym stanowisku dłuższym niż 3 lata lecz krótszym lub równym 5 lat nie znaleziono korelacji pomiędzy stażem na obecnym stanowisku a badanymi czynnikami satysfakcji.

Dodatkowo: znaleziono korelację pomiędzy stażem na obecnym stanowisku a stażem z obecnym managerem oraz stażem w firmie.

In [49]:
C4_corr = C4.corr()
C4_corr = C4_corr.style.applymap(corr_color)
C4_corr

Wśród osób o stażu nn obecnym stanowisku dłuższym niż 5 lata lecz krótszym lub równym 9 lat nie znaleziono korelacji pomiędzy stażem na obecnym stanowisku a badanymi czynnikami satysfakcji.

In [50]:
C5_corr = C5.corr()
C5_corr = C5_corr.style.applymap(corr_color)
C5_corr

Wśród osób o stażu nn obecnym stanowisku dłuższym niż 9 lat nie znaleziono korelacji pomiędzy stażem na obecnym stanowisku a badanymi czynnikami satysfakcji.

Dodatkowo: znaleziono korelacje pomiędzy stażem na obecnym stanowisku a miesięcznym przychodem i poziomem stanowiska.